trieste/models/utils.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module contains auxiliary objects and functions that are used by multiple model types.
+"""
+
+from __future__ import annotations
+
+import gpflow
+import tensorflow as tf
+from gpflow.utilities.traversal import _merge_leaf_components, leaf_components
+
+from .. import logging
+from ..data import Dataset
+from .interfaces import TrainableProbabilisticModel
+
+
+def write_summary_data_based_metrics(
+    dataset: Dataset,
+    model: TrainableProbabilisticModel,
```

Comment:
You don't need a trainable model here (and in fact the GPflowPredictor ABC isn't one itself).

```suggestion
    model: ProbabilisticModel,
```

Reply:
done